### PR TITLE
Add support for Python models

### PR DESCRIPTION
**Pull request by RaunoArike**
This pull request adds support for generating counterfactuals for machine learning models defined in Python using PyTorch. This greatly extends the scope of machine learning models the package is compatible with, as the model doesn't have to be written in Julia anymore. This is particularly important given the widespread use of Python in the machine learning community.

Our implementation uses `PythonCall.jl` for the conversion and is agnostic to model architecture: any model defined using PyTorch is supported. The PR also includes a report that describes our investigation into `ONNX.jl` for solving this problem, the reasoning behind our decision to use `PythonCall.jl` instead, and our general approach and design choices when implementing this conversion.

### Changes made

```
Implemented a basic interface for models defined using PyTorch.
Created infrastructure that enables the user to import any model that has been defined using PyTorch with its weights saved in a Pickle file.
Wrote methods that made the PyTorch models compatible with the counterfactual generators in our package.
Documented the whole process, including a report on our investigation into `ONNX.jl` and the design choices made.
```

### Testing

```
Tested simple PyTorch models on all synthetic datasets currently used in the package.
```

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -21,39 +21,43 @@ synthetic = _load_synthetic()
...
+    # @testset "Generators" begin
+    #     include("generators.jl")
+    # end
 
-    @testset "Probe" begin
-        include("probe.jl")
-    end
+    # @testset "Probe" begin
+    #     include("probe.jl")
+    # end
```
Comment (VincentPikand) [inline test/runtests.jl:46]: you know whats wrong

File: test/pytorch.jl [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+using CounterfactualExplanations.Generators
+using CounterfactualExplanations.Data
+using CounterfactualExplanations.Models
+using CounterfactualExplanations.DataPreprocessing
+using DataFrames
+using Flux
+using LinearAlgebra
+using MLUtils
+using Random
+using PythonCall
```
Comment (VincentPikand) [inline test/pytorch.jl:12]: Actually, all of these should be in the "base" file `runtests.jl`. Can you please open an issue for this?

Comment (VincentPikand) [reply test/pytorch.jl:12]: an Issue, because other files also don't follow this format

**Review by VincentPikand — CHANGES_REQUESTED** [review]
- structure the test files better, like the `data` folder

File: test/runtests.jl (hunk 1 above)
Comment (RaunoArike) [reply test/runtests.jl:46]: Yes, I'm making some late experiments with additional tests, will fix once the experiment is finished

File: test/pytorch.jl (hunk 2 above)
Comment (VincentPikand) [reply test/pytorch.jl:12]: no, not required

**Review by VincentPikand — APPROVED** [review]
You already anticipated my requested changes like a Yu-Gi-Oh! player, so LGTM. Remember to include all tests when you finish with your experimentation.

**Pull request merged**